src/main/scala/viper/gobra/frontend/InformativeErrorListener.scala

```diff
@@ -0,0 +1,246 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2022 ETH Zurich.
+
+package viper.gobra.frontend
+
+import org.antlr.v4.runtime.misc.IntervalSet
+import org.antlr.v4.runtime.{BaseErrorListener, CommonTokenStream, FailedPredicateException, InputMismatchException, Lexer, NoViableAltException, Parser, RecognitionException, Recognizer, Token}
+import org.bitbucket.inkytonik.kiama.util.{FileSource, Source}
+import viper.gobra.frontend.GobraParser.{CapContext, EosContext, ExpressionContext, ImplementationProofContext, RULE_blockWithBodyParameterInfo, RULE_eos, RULE_shortVarDecl, RULE_type_, RULE_varDecl, Slice_Context, TypeSpecContext, Type_Context, VarSpecContext, ruleNames}
+import viper.gobra.frontend.Source.FromFileSource
+import viper.gobra.reporting.ParserError
+import viper.silver.ast.SourcePosition
+
+import java.nio.file.Path
+import scala.collection.mutable.ListBuffer
+
+class InformativeErrorListener(val messages: ListBuffer[ParserError], val source: Source) extends BaseErrorListener {
```

Comment:
This file needs more documentation. Please add:

1. For your analyze... methods, please add a description of which case they are handling. E.g. that a superfluous token was found which was not expected, or that a token was expected, etc. 
2. Whenever a non-default error type is returned, then please add a sentence describing what is going on. This should be as unambiguous as possible. E.g. "Found ':=', but expected end of line. Return suggestion to use '=' ". 
3. Also add to all other method 1-2 sentences describing what they are doing.